users/policies/service.go

```diff
@@ -72,31 +73,35 @@ func (svc service) AddPolicy(ctx context.Context, token string, p Policy) error
 	if err != nil {
 		return err
 	}
-	p.UpdatedBy = id
 	if err := p.Validate(); err != nil {
 		return err
 	}
 
-	page, err := svc.policies.Retrieve(ctx, Page{Subject: p.Subject, Object: p.Object})
+	pm := Page{Subject: p.Subject, Object: p.Object, Offset: 0, Limit: 1}
+	page, err := svc.policies.Retrieve(ctx, pm)
 	if err != nil {
 		return err
 	}
+
+	// If the policy already exists, update it appending the new actions
```

Comment:
No need to make it complex, just replace actions with new ones. If someone wants to update the policy, they will use the update endpoint.

Reply:
Okay